### PR TITLE
[SPARK-26602] Insert into table fails after querying UDF loaded with wrong hdfs path

### PR DESCRIPTION
**Pull request by chakravarthiT**
## What changes were proposed in this pull request?
The problem here is, though the jar does not exist ,it is added to the addedJars in sparkContext.scala, when performing select on UDF. And when insert into table happens it is trying to load the jars from the ListJars and as the jar not exist,it gives Exception.

So in this PR,validation is added before adding the jars to the addedJars.

## How was this patch tested?
Manual Test

I have attached stacktrace below.
[beforeFixUdf.txt](https://github.com/apache/spark/files/2930184/beforeFixUdf.txt)
[afterFixUdf.txt](https://github.com/apache/spark/files/2930183/afterFixUdf.txt)

### CODE REVIEW

**Comment by chakravarthiT** [comment]
@srowen  Please review.

**Comment by srowen** [comment]
This sounds like a user error. I don't think Spark should work around it. Silently ignoring classpath entries that don't exist will cause later errors anyway that may be harder to debug.

**Comment by HyukjinKwon** [comment]
Yea, let's don't fix it.

File: core/src/main/scala/org/apache/spark/SparkContext.scala [hunk 1]
```diff
@@ -1814,12 +1814,16 @@ class SparkContext(config: SparkConf) extends Logging {
...
         val timestamp = System.currentTimeMillis
-        if (addedJars.putIfAbsent(key, timestamp).isEmpty) {
-          logInfo(s"Added JAR $path at $key with timestamp $timestamp")
-          postEnvironmentUpdate()
-        } else {
-          logWarning(s"The jar $path has been added already. Overwriting of added jars " +
-            "is not supported in the current version.")
+        //  SPARK-26602: Make sure the path exists,before adding jar to addedJars
+        val fs = FileSystem.get(hadoopConfiguration)
+        if (fs.exists(new Path(path))) {
```
Comment (HyukjinKwon) [inline core/src/main/scala/org/apache/spark/SparkContext.scala]: BTW, this looks going to fail if we use `local` scheme.

**Pull request closed without merging**